### PR TITLE
LPD-86315 Detect GPC Signal via Server

### PR DESCRIPTION
**Pull request by cdbm**
https://liferay.atlassian.net/browse/LPD-86315

**What is being fixed:** Liferay had no server-side way to detect the Global Privacy Control (GPC) `Sec-GPC` HTTP header. Without server-side detection, backend components cannot honor a visitor's opt-out signal before a page response is generated.

**How it is being fixed:** Introduces a new `GlobalPrivacyControlProvider` OSGi service exposed from the cookies module. The implementation gates on the `LPD-75064` feature flag, resolves the active scope (group when a layout is on the request, otherwise company), and consults `CookiesConfigurationProvider` to verify that cookies preference handling and Global Privacy Control are both enabled for that scope. Only then does it inspect the `Sec-GPC` request headers and return `true` when any header value equals `1` after trimming optional whitespace. An integration test covers the feature-flag gate, each configuration toggle, site-over-company override and fallback, system-PID fallback, and the header parsing edge cases (absent, zero, unrecognized token, comma-separated, whitespace, multiple header instances).

### CODE REVIEW

**Comment by cdbm** [comment]
ci:test:sf

**Comment by cdbm** [comment]
ci:test:sf

File: modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/provider/GlobalPrivacyControlProviderImpl.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		if (_cookiesConfigurationProvider.isCookiesPreferenceHandlingEnabled(
+				scope, scopePK) &&
+			_cookiesConfigurationProvider.
+				isCookiesPreferenceHandlingGlobalPrivacyControlEnabled(
+					scope, scopePK)) {
+
+			return true;
+		}
+
+		return false;
```
[the remark this answers is not in the document]
Comment (cdbm) [reply modules/apps/cookies/cookies-impl/src/main/java/com/liferay/cookies/internal/configuration/provider/GlobalPrivacyControlProviderImpl.java:67]: Tried this — Liferay's source formatter (`SourceCheck:JavaReturnStatementCheck`) reverts it back to the `if (X) { return true; } return false;` form when the boolean expression spans multiple lines, so the simpler `return X;` won't survive `ant format-source-current-branch`. Leaving as-is.

File: modules/apps/cookies/cookies-api/src/main/java/com/liferay/cookies/configuration/GlobalPrivacyControlProvider.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2026 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.cookies.configuration;
```
Comment (rafaprax) [inline modules/apps/cookies/cookies-api/src/main/java/com/liferay/cookies/configuration/GlobalPrivacyControlProvider.java:6]: it is not in the right package

File: modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java [hunk 3]
```diff
@@ -0,0 +1,371 @@
...
+		if (_group != null) {
+			GroupTestUtil.deleteGroup(_group);
+
+			_group = null;
+			_layout = null;
+		}
+	}
+
+	@Test
+	public void testConsentManagerDisabled() throws Exception {
```
Comment (rafaprax) [inline modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java:81]: try to merge the tests and call the method `testIsSignalActive`

File: modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java [hunk 4]
```diff
@@ -0,0 +1,371 @@
...
+
+		MockHttpServletRequest mockHttpServletRequest = _createRequest(
+			secGpcHeader);
+
+		mockHttpServletRequest.setAttribute(WebKeys.LAYOUT, _layout);
+
+		return mockHttpServletRequest;
+	}
+
+	private MockHttpServletRequest _createRequest(String secGpcHeader) {
```
Comment (rafaprax) [inline modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java:289]: rename to `_createMockHttpServletRequest`

File: modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java [hunk 5]
```diff
@@ -0,0 +1,371 @@
...
+		MockHttpServletRequest mockHttpServletRequest =
+			new MockHttpServletRequest();
+
+		mockHttpServletRequest.setAttribute(WebKeys.COMPANY_ID, _companyId);
```
Comment (rafaprax) [inline modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java:293]: it shoulb be after the `mockHttpServletRequest.addHeader` call

File: modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java [hunk 6]
```diff
@@ -0,0 +1,371 @@
...
+
+		ConfigurationTestUtil.saveConfiguration(
+			_SYSTEM_PID,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"enabled", cookiesPreferenceHandlingEnabled
+			).put(
+				"globalPrivacyControlEnabled", globalPrivacyControlEnabled
+			).build());
+
+		_systemConfigurationSaved = true;
```
Comment (rafaprax) [inline modules/apps/cookies/cookies-test/src/testIntegration/java/com/liferay/cookies/internal/configuration/provider/test/GlobalPrivacyControlProviderImplTest.java:351]: get rid of this boolean variable. Store the result of the saveConfiguration into a variable like  `_groupConfigurationPid`. Maybe `_systemConfigurationPid`?

**Review by rafaprax — CHANGES_REQUESTED** [review]
merge it with the https://github.com/liferay-appsec/liferay-portal/pull/2783
